warpy-ai/tgs
# 1 set up the basic rust project structure

#### **Description**:
This PR introduces the foundational structure for the TGS project. It sets up a basic Rust project with a simple shell loop that reads user input and echoes it back to the user. This serves as a starting point for integrating conversational capabilities and other advanced features.

#### **Changes**:
1. Initialized a new Rust project within the `tgs/` directory.
2. Implemented a basic shell loop in `main.rs` that:
   - Prompts the user with `tgs>`.
   - Reads user input.
   - Echoes back the entered input with a "You entered:" prefix.
3. Added a `.gitignore` file to the root directory to handle both Node.js and Rust-related ignores.

#### **Testing**:
1. Navigate to the `tgs/` directory.
2. Run `cargo run`.
3. Enter any text at the `tgs>` prompt and observe the echoed response.

#### **Next Steps**:
- Integrate the NLP module to process conversational input.
- Implement error handling and other core shell functionalities.
- Enhance the shell loop with context awareness and command history.

#### **Screenshots**:
_(Optional: Attach any relevant screenshots or GIFs showcasing the changes.)_

---

**Reviewer Checklist**:
- [x] Code structure and organization.
- [x] Functionality matches the described changes.
- [x] No regression issues.
- [x] Adequate inline documentation and comments.
- [x] Relevant tests and their passing status.

---

Once you've filled in any project-specific details or made modifications as needed, you can use this text when creating your pull request on your version control platform (e.g., GitHub, GitLab, Bitbucket).
